tower-http/src/services/fs/serve_dir.rs

```diff
@@ -329,14 +336,35 @@ impl<ReqBody> Service<Request<ReqBody>> for ServeDir {
 
             match request_method {
                 Method::HEAD => {
-                    let (metadata, maybe_encoding) =
+                    let (meta, maybe_encoding) =
                         file_metadata_with_fallback(full_path, negotiated_encodings).await?;
-                    Ok(Output::FileMetadata(metadata, mime, maybe_encoding))
+                    let maybe_range = try_parse_range(range_header.as_ref(), meta.len());
```

Comment:
I think it would be odd for a client to send request headers in a HEAD request but I guess it doesn't hurt to support

Reply:
It's kind of odd but per our discussion, head requests should return the same headers as a get without the body, and we only check this if the client actual requests range in their HEAD

Reply:
To it feels like something we should support